modules/meilisearch/meilisearch.go

```diff
@@ -0,0 +1,119 @@
+package meilisearch
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	defaultMasterKey = "just-a-master-key-for-test"
+	defaultHTTPPort  = "7700/tcp"
+	masterKeyEnvVar  = "MEILI_MASTER_KEY"
+)
+
+// MeilisearchContainer represents the Meilisearch container type used in the module
+type MeilisearchContainer struct {
+	testcontainers.Container
+	MasterKey string
+}
+
+// Run creates an instance of the Meilisearch container type
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*MeilisearchContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{defaultHTTPPort},
+		Env: map[string]string{
+			masterKeyEnvVar: defaultMasterKey,
+		},
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	// Gather all config options (defaults and then apply provided options)
+	settings := defaultOptions()
+	for _, opt := range opts {
+		if apply, ok := opt.(Option); ok {
+			apply(settings)
+		}
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, err
+		}
+	}
+
+	if settings.DumpDataFilePath != "" {
+		genericContainerReq.Files = []testcontainers.ContainerFile{
+			{
+				HostFilePath:      settings.DumpDataFilePath,
+				ContainerFilePath: "/dumps/" + settings.DumpDataFileName,
```

Comment:
@mashail do you think we need to validate the dump files have the `.dump` extension? At least, I think we should include the format in the docs, pointing to the official docs. Wdyt?

Reply:
I went ahead and added it to the docs. In the end, the users using dumps should know the proper format.

Thanks!